sdk/messaging/azservicebus/internal/namespace.go

```diff
@@ -442,16 +436,35 @@ func (ns *Namespace) getUserAgent() string {
 	return userAgent
 }
 
-func (ns *Namespace) getResourceURI() string {
-	if ns.Environment.ResourceIdentifiers.ServiceBus == "" {
-		return serviceBusResourceURI
-	}
-	return ns.Environment.ResourceIdentifiers.ServiceBus
-}
-
 func (ns *Namespace) resolveSuffix() string {
 	if ns.Suffix != "" {
 		return ns.Suffix
 	}
 	return azure.PublicCloud.ServiceBusEndpointSuffix
 }
+
+// nextClaimRefreshDuration figures out the proper interval for the next authorization
+// refresh.
+//
+// It applies a few real world adjustments:
+// - We assume the expiration time is 10 minutes ahead of when it actually is, to adjust for clock drift.
+// - We don't let the refresh interval fall below 2 minutes
+// - We don't let the refresh interval go above 49 days
+//
+// This logic is from here:
+// https://github.com/Azure/azure-sdk-for-net/blob/bfd3109d0f9afa763131731d78a31e39c81101b3/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs#L998
+func nextClaimRefreshDuration(expirationTime time.Time, currentTime time.Time) time.Duration {
+	const min = 2 * time.Minute
+	const max = 49 * 24 * time.Hour
+	const clockDrift = 10 * time.Minute
+
+	var refreshDuration = expirationTime.Sub(currentTime) - clockDrift
+
+	if refreshDuration < min {
+		return min
+	} else if refreshDuration > max {
+		return max
+	} else {
```

Comment:
Nit, you can drop the `else` and just `return refreshDuration`

Reply:
Done